cmd/main.go

```diff
@@ -43,33 +44,34 @@ func main() {
 	isArgocdCLI := false
 
 	switch binaryName {
-	case "argocd", "argocd-linux-amd64", "argocd-darwin-amd64", "argocd-windows-amd64.exe":
+	case common.CLI:
 		command = cli.NewCommand()
 		isArgocdCLI = true
-	case "argocd-server":
+	case common.Server:
 		command = apiserver.NewCommand()
-	case "argocd-application-controller":
+	case common.ApplicationController:
 		command = appcontroller.NewCommand()
-	case "argocd-repo-server":
+	case common.RepoServer:
 		command = reposerver.NewCommand()
-	case "argocd-cmp-server":
+	case common.CMPServer:
 		command = cmpserver.NewCommand()
 		isArgocdCLI = true
-	case "argocd-commit-server":
+	case common.CommitServer:
 		command = commitserver.NewCommand()
-	case "argocd-dex":
+	case common.Dex:
 		command = dex.NewCommand()
-	case "argocd-notifications":
+	case common.Notifications:
 		command = notification.NewCommand()
-	case "argocd-git-ask-pass":
+	case common.GitAskPass:
 		command = gitaskpass.NewCommand()
 		isArgocdCLI = true
-	case "argocd-applicationset-controller":
+	case common.ApplicationSetController:
 		command = applicationset.NewCommand()
-	case "argocd-k8s-auth":
+	case common.K8sAuth:
 		command = k8sauth.NewCommand()
 		isArgocdCLI = true
 	default:
+		// "argocd-linux-amd64", "argocd-darwin-amd64", "argocd-windows-amd64.exe" are also valid binary names
```

Comment:
```suggestion
		// "argocd-linux-amd64", "argocd-darwin-amd64", "argocd-windows-amd64.exe" are also valid binary names
```
Interesting. You can ignore [this comment then](https://github.com/argoproj/argo-cd/pull/24486/files#r2336253434).